receiver/telemetrygeneratorreceiver/config.go

```diff
@@ -105,6 +115,62 @@ func validateLogGeneratorConfig(g *GeneratorConfig) error {
 	return nil
 }
 
+func validateOTLPGenerator(cfg *GeneratorConfig) error {
+
+	telemetryType, ok := cfg.AdditionalConfig["telemetry_type"]
+	if !ok {
+		return errors.New("telemetry_type must be set")
+	}
+
+	// validate the telemetry type
+	telemetryTypeStr, ok := telemetryType.(string)
+	if !ok {
+		return fmt.Errorf("invalid telemetry type: %v", telemetryType)
+	}
+	dataType := component.DataType(telemetryTypeStr)
+	switch dataType {
+	case component.DataTypeLogs, component.DataTypeMetrics, component.DataTypeTraces:
+	default:
+		return fmt.Errorf("invalid telemetry type: %s", telemetryType)
+	}
+
+	// validate the otlp json
+	otlpJSON, ok := cfg.AdditionalConfig["otlp_json"]
+	if !ok {
+		return errors.New("otlp_json must be set")
+	}
+
+	otlpJSONStr, ok := otlpJSON.(string)
+	if !ok {
+		return fmt.Errorf("otlp_json must be a string, got: %v", otlpJSON)
+	}
+
+	jsonBytes := []byte(otlpJSONStr)
+
+	switch dataType {
+	case component.DataTypeLogs:
+		marshaler := plog.JSONUnmarshaler{}
+		_, err := marshaler.UnmarshalLogs(jsonBytes)
+		if err != nil {
+			return fmt.Errorf("error unmarshalling otlp logs json: %w", err)
+		}
+	case component.DataTypeMetrics:
+		marshaler := pmetric.JSONUnmarshaler{}
+		_, err := marshaler.UnmarshalMetrics(jsonBytes)
+		if err != nil {
+			return fmt.Errorf("error unmarshalling otlp metrics json: %w", err)
+		}
+	case component.DataTypeTraces:
+		marshaler := ptrace.JSONUnmarshaler{}
+		_, err := marshaler.UnmarshalTraces(jsonBytes)
+		if err != nil {
+			return fmt.Errorf("error unmarshalling otlp traces json: %w", err)
+		}
+
+	}
```

Comment:
So for these I noticed if I put my `telemetry_type` as `logs` and my `otlp_json` as metrics I don't get an error. It seems to just try to emit an empty payload.

Reply:
Good catch, added validation